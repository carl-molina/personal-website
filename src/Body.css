.Kett, .RS, h4 a {
  text-decoration: none;
  transition: background-color 0.33s ease;
}

.Kett:hover {
  text-decoration: none;
  background-color: rgb(61, 129, 112);
}

.RS:hover {
  background-color: rgb(61, 129, 112);
}
