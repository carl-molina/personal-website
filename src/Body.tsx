import React from "react";


/** Body: presentational component for about me details.
 *
 *  Props: none
 *  State: none
 *
 *  App -> Body
 */

function Body() {

    return (
        <>
            <div className="Body" id="about-me">
                <h2>Film, Autonomous Vehicles, Software Engineering</h2>
            </div>
        </>
    )
}

export default Body;
