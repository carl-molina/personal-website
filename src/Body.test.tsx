import React from "react";
import { test, expect, describe } from "vitest";
import { render } from '@testing-library/react';

import Body from './Body';


describe("Body smoke test", function () {
  test('renders without crashing', function () {
      render(<Body />);
  });
});

describe('Body renders correct data', function () {
  test('contains expected text', function () {
      const res = render(<Body />);
      expect(res.queryByText('Experience')).toBeInTheDocument();
      expect(res.queryByText('Education')).toBeInTheDocument();
      expect(res.queryByText('Bachelor of Arts, Screenwriting')).toBeInTheDocument();
  });
});

describe("Body snapshot test", function () {
  test('matches snapshot', function () {
      const { container } = render(<Body />);
      expect(container).toMatchSnapshot();
  });
});
