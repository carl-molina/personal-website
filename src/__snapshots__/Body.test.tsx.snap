// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Projects snapshot test > matches snapshot 1`] = `
<div>
  <div
    class="Body-container"
  >
    <h2
      class="Body-header mt-5"
    >
      Experience
    </h2>
    <h3>
      Field Operator Specialist III
    </h3>
    <h4
      class="Kett"
    >
      <a
        href="https://www.ketteng.com/"
        rel="noopener"
        target="_blank"
      >
        Kett Engineering 2018-2023
      </a>
    </h4>
    <h2
      class="Body-header mt-5"
    >
      Education
    </h2>
    <h3
      class="RS-title mt-4"
    >
      Full Stack Web Development
    </h3>
    <h4
      class="RS"
    >
      <a
        href="https://www.rithmschool.com/"
        rel="noopener"
        target="_blank"
      >
        Rithm School 2023-2024
      </a>
    </h4>
    <h3
      class="De-Anza mt-4"
    >
      Associate of Science, Computer Science
    </h3>
    <h4
      class="RS"
    >
      <a
        href="https://www.deanza.edu/"
        rel="noopener"
        target="_blank"
      >
        De Anza College 2019
      </a>
    </h4>
    <h3
      class="LMU mt-4"
    >
      Bachelor of Arts, Screenwriting
    </h3>
    <h4
      class="RS"
    >
      <a
        href="https://www.lmu.edu/"
        rel="noopener"
        target="_blank"
      >
        Loyola Marymount University 2016
      </a>
    </h4>
  </div>
</div>
`;
